webapp/store/middleware.js

```diff
@@ -0,0 +1,34 @@
+import { getClient } from '@bpanel/bpanel-utils';
+
+export function errorCatcherMiddleware(errorHandler) {
+  return function(store) {
+    return function(next) {
+      return function(action) {
+        try {
+          return next(action);
+        } catch (err) {
+          const message = `There was an error in the middleware for the action ${
+            action.type
+          }: `;
+          errorHandler(message, err, store.getState, action, store.dispatch);
+          return err;
+        }
+      };
+    };
+  };
+}
+
+export function clientMiddleware() {
+  return function(next) {
+    return function(action) {
+      const client = getClient();
+      if (action.type === 'SET_CURRENT_CLIENT') {
+        const clientInfo = action.payload;
+        const { id, chain = 'bitcoin' } = clientInfo;
```

Comment:
I feel like having `bitcoin` hardcoded in here could lead to developer confusion or a bug in the future. Maybe add a big comment to make it easier to explain that it defaults to `bitcoin`

Reply:
Yeah, this is tough. We do something similar on the server end with configs that are not explicit. Will add a comment to help clarify.